### PR TITLE
Adds wall-attached defibrillator mounts

### PR DESCRIPTION
**Pull request by ghost**
:cl: Xhuis
add: Defibrillator mounts can now be found in the CMO's locker or produced from a medical protolathe after Biotech is unlocked. They can be attached to walls, and hold defibrillators for public use. You can swipe an ID to clamp the defibrillator in place, and it will automatically draw from the powernet to recharge it. Being one tile away from the mount will force you to drop the paddles.
/:cl:

Adds defib mounts. They can be installed through unhooked item versions that are found in the CMO's locker (1) and created from the medical protolathe. They:
* Hold defibrillators for nearby people to use; you can interact with the mount to grab the paddles. Walking at least one tile away will cause the paddles to snap back. Alt-clicking will remove the defib.
* Draw from the powernet to recharge defibrillators at about half the speed of a cell charger, without having to remove the defibrillator.
* Can be locked in place by swiping an ID with medical, security, or head-of-staff access to prevent people from taking them. You can override these clamps with a multitool after a chunk of time in case of an emergency.
  * The clamps also become unlockable by anyone during red alert.

I made this since defibrillators are something I often see lying around for people to make use of, but it's awkward to get it onto your back and then off again afterwards and because anyone can just take it and walk off, depriving you of a defib.

### CODE REVIEW

**Comment by epochayur** [comment]
top-tier PR, this is about 8000 times better than leaving them on a table for some greyshirt to wander off with

**Comment by Spaceinaba** [comment]
FINALLY

**Comment by nicbn** [comment]
Oh no I hate these things I always miss them and think that someone took the defibs

**Comment by DaxDupont** [comment]
Can you make it so emag unlocks it too?

**Comment by Xhuis** [comment]
Emags are already pretty massively powerful for what they are, and I've seen maintainers state that they want them to be less so, so I think it's fine as-is. Multitools are pretty easy to get.

**Comment by calzilla1** [comment]
Nake it so that if you emag it you can shoot tesla bolts out of it

**Comment by KieranCat** [comment]
jesus christ satan calm down. If it is implemented, make it have a massive power drain on the network. But yes, +1! Been waiting for defibs like this for a while now.

**Comment by Xhuis** [comment]
It drains a bit less power than putting the defib's cell in a charger would, using around 100 W per second from the powernet to recharge 90 W per second in the power cell (cell chargers are 200 W per second for 175 W). This does technically mean that defib mounts can be used as ghetto power cell chargers, which I'm fine with.

File: code/game/objects/structures/crates_lockers/closets/secure/medical.dm [hunk 1]
```diff
@@ -74,6 +74,8 @@
 	new /obj/item/door_remote/chief_medical_officer(src)
 	new /obj/item/clothing/neck/petcollar(src)
 	new /obj/item/pet_carrier(src)
+	for(var/i in 1 to 3)
+		new /obj/item/wallframe/defib_mount(src)
```
Comment (Cyberboss) [inline code/game/objects/structures/crates_lockers/closets/secure/medical.dm]: I'm going to go out on a limb here and say only put one of these here so they have a reason to actually be made at the lathe

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by 81Denton** [comment]
Shouldn't it drain more power since it saves you the step of removing the cell and putting it in a charger
I'm just nitpicking, not like power currently matters anyway

**Review by Cyberboss — DISMISSED** [review]
Add a get_cell() override which calls the defib's get_cell(). Also make sure things are handled gracefully if the wall_mount is deleted, destroyed, or deconned while holding the paddles

**Comment by Xhuis** [comment]
Hey, sorry, I missed your comment. I'll get some work done on this shortly and squash commits.

**Comment by Xhuis** [comment]
nani?!

**Comment by Xhuis** [comment]
Any maintainers that aren't Cyberboss, you guys can go home, don't worry. I accidentally pulled from master and botched the commits, but it's cleaned up now.

File: code/game/machinery/defibrillator_mount.dm [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+	var/obj/item/defibrillator/defib //this mount's defibrillator
+	var/clamps_locked = FALSE //if true, and a defib is loaded, it can't be removed without unlocking the clamps
+
+/obj/machinery/defibrillator_mount/loaded/Initialize() //loaded subtype for mapping use
+	. = ..()
+	defib = new/obj/item/defibrillator/loaded(src)
+
+/obj/machinery/defibrillator_mount/Destroy()
+	if(defib)
+		if(defib.paddles)
```
Comment (AnturK) [inline code/game/machinery/defibrillator_mount.dm]: Shouldn't this be handled by the item qdel ?

File: code/game/machinery/defibrillator_mount.dm [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+		if(I.flags_1 & NODROP_1 || !user.transferItemToLoc(I, src))
+			to_chat(user, "<span class='warning'>[I] is stuck to your hand!</span>")
+			return
+		user.visible_message("<span class='notice'>[user] hooks up [I] to [src]!</span>", \
+		"<span class='notice'>You press [I] into the mount, and it clicks into place.</span>")
+		playsound(src, 'sound/machines/click.ogg', 50, TRUE)
+		defib = I
+		update_icon()
+		return
+	else if(istype(I, /obj/item/card/id) || istype(I, /obj/item/device/pda) || istype(I, /obj/item/device/modular_computer/tablet))
```
Comment (AnturK) [inline code/game/machinery/defibrillator_mount.dm]: Isn't there a helper for this ?

File: code/game/machinery/defibrillator_mount.dm [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+		return
+	user.visible_message("<span class='notice'>[user] pulses [multitool], and [src]'s clamps slide up.</span>", \
+	"<span class='notice'>You override the locking clamps on [src]!</span>")
+	playsound(src, 'sound/machines/locktoggle.ogg', 50, TRUE)
+	clamps_locked = FALSE
+	update_icon()
+	return TRUE
+
+/obj/machinery/defibrillator_mount/AltClick(mob/living/user)
+	if(!user.Adjacent(src))
```
Comment (AnturK) [inline code/game/machinery/defibrillator_mount.dm]: Also needs istype.

**Review by AnturK — CHANGES_REQUESTED** [review]
Some minor stuff.

**Comment by AnturK** [comment]
Note to past self : don't review past 4 am

File: code/modules/research/designs.dm [hunk 5]
```diff
@@ -19,14 +19,14 @@ The currently supporting non-reagent materials. All material amounts are set as
...
-Design Guidlines
+Design Guidelines
 - When adding new designs, check rdreadme.dm to see what kind of things have already been made and where new stuff is needed.
 - A single sheet of anything is 2000 units of material. Materials besides metal/glass require help from other jobs (mining for
 other types of metals and chemistry for reagents).
 - Add the AUTOLATHE tag to
 */
 
-//DESIGNS ARE GLOBAL. DO NOT CREATE OR DESTROY THEM AT RUNTIME OUTSIDE OF INIT, JUST REFERENCE THEM TO WHATEVER YOU'RE DOING!
+//DESIGNS ARE GLOBAL. DO NOT CREATE OR DESTROY THEM AT RUNTIME OUTSIDE OF INIT, JUST REFERENCE THEM TO WHATEVER YOU'RE DOING! //why are you yelling?
```
Comment (MrStonedOne) [inline code/modules/research/designs.dm]: @duncathan

**Pull request merged**